src/main/scala/za/co/absa/hyperdrive/trigger/api/rest/services/WorkflowService.scala

```diff
@@ -82,8 +82,12 @@ class WorkflowServiceImpl(override val workflowRepository: WorkflowRepository,
 
   override def updateWorkflow(workflow: WorkflowJoined)(implicit ec: ExecutionContext): Future[Either[Seq[ApiError], WorkflowJoined]] = {
     val userName = getUserName.apply();
+    var originalWorkflow: WorkflowJoined = null;
+      getWorkflow(workflow.id)
+      .map(workflow => originalWorkflow = workflow);
```

Comment:
This is really dangerous.
Block of code bellow can be executed before this (this part will be never executed).
You can include getWorkflow into for bellow.
for {
      originalWorkflow <- getWorkflow(workflow.id)
....